pkg/filesystem/device_linux.go

```diff
@@ -0,0 +1,39 @@
+// +build linux
+
+package filesystem
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/buildbarn/bb-storage/pkg/filesystem"
+	"github.com/buildbarn/bb-storage/pkg/util"
+)
+
+// Device stores the name, device number and mode of a device to be used
+// when creating device nodes in the input root dev directory
+type Device struct {
+	Name string
+	Rdev uint64
+}
+
+// CreateDev will take a directory and a list of devices and make nodes
+// of those devices in the /dev subdirectory of the given directory
+func CreateDev(inputRootDirectory filesystem.Directory, devices []Device) error {
+	if err := inputRootDirectory.Mkdir("dev", 0777); err != nil && !os.IsExist(err) {
+		return util.StatusWrap(err, "Unable to create dev directory in input root")
+	}
+	devDir, err := inputRootDirectory.EnterDirectory("dev")
+	if err != nil {
+		return util.StatusWrap(err, "Unable to enter dev directory in input root")
+	}
+	for _, device := range devices {
+		if _, err := devDir.Lstat(device.Name); err == nil {
+			continue
+		}
+		if err := devDir.Mknod(device.Name, os.FileMode(syscall.S_IFCHR|0666), int(device.Rdev)); err != nil {
```

Comment:
So this is just about getting character devices working, right? Not block devices. Do we care about block devices?

If we don't care about block devices, could we change the phrasing everywhere to use 'character device' instead of just 'device'? If we do care about block devices, then we should let main.go save the file type as well.

Note that I'm fine with just supporting character devices. If we limit ourselves to that, main.go should have a check to enforce that as well.

Reply:
I believe we do only care about character devices in this case, so I'll go about making that more explicit throughout.